src/main/groovy/wooga/gradle/version/internal/VersionCode.groovy

```diff
@@ -147,4 +148,16 @@ class VersionCode {
     static Integer generateBuildNumberVersionCode(GitVersionRepository versionRepo, Boolean countPrerelease = true) {
         return versionRepo.countReachableVersions(countPrerelease) + 1
     }
+
+    /**
+     * Returns the time since 1.1.2024 in seconds
+     * should fit in 31 bits (we need one for the sign) for about 68 years until 2092
+     * @return
+     */
+    static Integer generateBuildTimeVersionCode() {
+        def date = new Date()
+        def buildTime = date.getTime() - 1704063600000 // 1.1.2024
```

Comment:
you can use `System.currentTimeMillis()` instead of `new Date().getTime()`.

Also, is there a special reason for the timestamp to start at 1.1.24? It is expected that timestamps go with Unix Time, so this may be very confusing in the future.

Reply:
Yes. I hinted to it in the comment. It has to fit in 31 bits.